lib/team_processor.rb

```diff
@@ -0,0 +1,133 @@
+module TeamProcessor
+  def season_stats(team_id, game_teams)
+    team_seasons = Hash.new { |season_id, games_won| season_id[games_won] = [0.0, 0.0] }
+    game_teams.each do |game_team|
+      if game_team.team_id == team_id
+        team_seasons[game_team.game_id[0..3]][0] += 1
+        if game_team.result == 'WIN'
+          team_seasons[game_team.game_id[0..3]][1] += 1
+        end
+      end
+    end
+    # require "pry";binding.pry
+    return team_seasons
+  end
+
+  def bestest_season(team_seasons)
+    highest_win_percentage = 0.0
+    winningest_season = ''
+    team_seasons.each do |season, wins_games|
+      if highest_win_percentage < wins_games[1] / wins_games[0]
+        highest_win_percentage = wins_games[1] / wins_games[0]
+        winningest_season = season
+      end
+    end
+    return winningest_season
+  end
+
+  def worstest_season(team_seasons)
+    lowest_win_percentage = 1.0
+    losingest_season = ''
+    team_seasons.each do |season, wins_games|
+      if lowest_win_percentage > wins_games[1] / wins_games[0]
+        lowest_win_percentage = wins_games[1] / wins_games[0]
+        losingest_season = season
+      end
+    end
+    return losingest_season
+  end
+
+  def games_played(team_id, game_teams)
+    games_played = 0.0
+    game_teams.each do |game_team|
+      if game_team.team_id == team_id
+        games_played += 1
+      end
+    end
+    games_played
+  end
+
+  def games_won(team_id, game_teams)
+    games_won = 0.0
+    game_teams.each do |game_team|
+      if game_team.team_id == team_id
+        if game_team.result == "WIN"
+          games_won += 1
+        end
+      end
+    end
+    games_won
+  end
+
+  def goals_scored(team_id, value, game_teams)
+    if value == "most"
+      highest_goal = 0
+      game_teams.each do |game_team|
+        if game_team.team_id == team_id
+          if game_team.goals.to_i > highest_goal
+            highest_goal = game_team.goals.to_i
+          end
+        end
+      end
+      return highest_goal
+    else
+      lowest_goal = Float::INFINITY
+      game_teams.each do |game_team|
+        if game_team.team_id == team_id
+          if game_team.goals.to_i < lowest_goal
+            lowest_goal = game_team.goals.to_i
+          end
+        end
+      end
+      return lowest_goal
+    end
+  end
+
+  def opponent_stats(team_id, games)
```

Comment:
i like the use of opponent_stats here in your module. it allows to be used to both rival and favorite opponent!